integrations/aws/aws/session_manager.py

```diff
@@ -0,0 +1,161 @@
+            )
+
+            credentials = organizations_client["Credentials"]
+            organization_role_session = aioboto3.Session(
+                aws_access_key_id=credentials["AccessKeyId"],
+                aws_secret_access_key=credentials["SecretAccessKey"],
+                aws_session_token=credentials["SessionToken"],
+            )
+
+            return organization_role_session
+
+    def _get_account_read_role_name(self) -> str:
+        return ocean.integration_config.get("account_read_role_name", "")
+
+    async def _update_available_access_credentials(self) -> None:
+        logger.info("Updating AWS credentials")
+        if not self._application_session or not self._organization_reader:
+            raise ValueError(
+                "Application session or organization reader session not initialized yet"
+            )
+
+        async with (
+            self._application_session.client("sts") as sts_client,
+            self._organization_reader.client("organizations") as organizations_client,
+        ):
+            paginator = organizations_client.get_paginator("list_accounts")
+            try:
+                async for page in paginator.paginate():
+                    for account in page["Accounts"]:
+                        if account["Id"] == self._application_account_id:
+                            # Skip the current account as it is already added
+                            # Replace the Temp account details with the current account details
+                            self._aws_accessible_accounts[0] = account
+                            continue
+                        await self._assume_role_and_update_credentials(
+                            sts_client, account
+                        )
+            except organizations_client.exceptions.AccessDeniedException:
+                logger.error(
+                    "Caller is not a member of an AWS organization. Assuming role in the current account."
+                )
+        logger.info(f"Found {len(self._aws_credentials)} AWS accounts")
+
+    async def _assume_role_and_update_credentials(
+        self, sts_client: STSClient, account: dict[str, Any]
+    ) -> None:
+        try:
+            account_role = await sts_client.assume_role(
+                RoleArn=f'arn:aws:iam::{account["Id"]}:role/{self._get_account_read_role_name()}',
+                RoleSessionName="AssumeRoleSession",
+            )
+            raw_credentials = account_role["Credentials"]
+            credentials = AwsCredentials(
+                account_id=account["Id"],
+                access_key_id=raw_credentials["AccessKeyId"],
+                secret_access_key=raw_credentials["SecretAccessKey"],
+                session_token=raw_credentials["SessionToken"],
+            )
+            await credentials.update_enabled_regions()
+            self._aws_credentials.append(credentials)
+            self._aws_accessible_accounts.append(account)
```

Comment:
Lets rethink the way you handle variables in this file to make this file more readable